examples/compose-replication-semisync.yml

```diff
@@ -0,0 +1,67 @@
+version: "3"
+x-common-variables: &common-variables
+  MARIADB_ROOT_PASSWORD: ${MARIADB_ROOT_PASSWORD}
+  MYSQL_INITDB_SKIP_TZINFO: ${MYSQL_INITDB_SKIP_TZINFO}
+  MARIADB_REPLICATION_USER: ${MARIADB_REPLICATION_USER}
+  MARIADB_REPLICATION_PASSWORD: ${MARIADB_REPLICATION_PASSWORD}
+
+x-common-attributes: &common-attributes
+  image: mariadb:lts
+  env_file:
+    - .env
+  environment: *common-variables
+  healthcheck:
+    test: ["CMD", "healthcheck.sh", "--connect", "--innodb_initialized"]
+    interval: 10s
+    timeout: 5s
+    retries: 3
+  logging:
+    driver: journald
+  networks:
+    - backend
+
+x-common-replication: &common-replication
+  healthcheck:
+    test: ["CMD", "healthcheck.sh", "--connect", "--replication_io", "--replication_sql", "--replication_seconds_behind_master=1", "--replication"]
+    interval: 10s
+    timeout: 5s
+    retries: 3
```

Comment:
is [start_period](https://docs.docker.com/compose/compose-file/compose-file-v3/#healthcheck) worth including and works correctly?10s?